src/Symfony/Component/Notifier/Bridge/Lox24/Lox24Transport.php

```diff
@@ -0,0 +1,187 @@
+        $body = $this->setVoiceLang($body, $options);
+
+        $response = $this->client->request('POST', sprintf('https://%s/sms', $this->getEndpoint()), [
+            'headers' => [
+                'X-LOX24-AUTH-TOKEN' => sprintf('%s:%s', $this->user, $this->token),
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/json',
+                'User-Agent' => 'LOX24 Symfony Notifier',
+            ],
+            'body' => $body,
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the remote LOX24 server.', $response, 0, $e);
+        }
+
+        if (201 !== $statusCode) {
+            $error = $response->toArray(false);
+
+            throw new TransportException(sprintf('Unable to send the SMS: "%s".', $error['detail']), $response);
+        }
+
+        $success = $response->toArray(false);
+
+        $sentMessage = new SentMessage($message, (string) $this);
+        $sentMessage->setMessageId($success['uuid']);
+
+        return $sentMessage;
+    }
+
+    private function isFromValid(string $from): bool
+    {
+        return preg_match('/^[.\-a-zA-Z0-9_ ]{2,11}$/', $from) || preg_match('/^\+[1-9]\d{1,14}$/', $from);
+    }
+
+    private function setIsTextDeleted(array $body, array $options): array
+    {
+        $body['is_text_deleted'] = (bool) ($options['is_text_delete'] ?? false);
+
+        return $body;
+    }
+
+    private function setCallbackData(array $body, array $options): array
+    {
+        if (!empty($options['callback_data'])) {
+            $body['callback_data'] = $options['callback_data'];
+        }
+
+        return $body;
+    }
+
+    private function setDeliveryAt(array $body, array $options): array
+    {
+        $body['delivery_at'] = max((int) ($options['delivery_at'] ?? 0), 0);
+
+        return $body;
+    }
+
+    private function setServiceCode(array $body, array $options): array
+    {
+        $code = $options['type'] ?? Type::Sms->value;
+        $type = Type::tryFrom((string) $code);
+
+        if (!$type) {
+            throw new InvalidArgumentException(sprintf('Invalid type: "%s".', $code));
+        }
+
+        $body['service_code'] = $type->getServiceCode();
+
+        return $body;
+    }
+
+    private function setVoiceLang(array $body, array $options): array
+    {
+        $voiceLang = $options['voice_lang'] ?? null;
+        if ($voiceLang) {
+            $voiceLang = strtolower($voiceLang);
+            if (!VoiceLanguage::tryFrom($voiceLang)) {
+                $allowed = implode(', ', array_map(static fn ($case) => $case->value, VoiceLanguage::cases()));
+                $str = 'The "voice_lang" option "%s" is not a valid language. Allowed languages are: %s.';
+                throw new InvalidArgumentException(sprintf($str, $voiceLang, $allowed));
+            }
```

Comment:
also use a try catch here?

Reply:
fixed

Reply:
@smnandre and @OskarStark please discuss how i should implement this code with `tryFrom()` or `from()` and try/catch.